app/lib/core/services/navigation/nav_service.dart

```diff
@@ -9,7 +9,12 @@ class NavigationService {
     return _navigationKey.currentState.pop();
   }
 
-  Future<dynamic> navigateTo(String routeName) {
-    return _navigationKey.currentState.pushNamed(routeName);
+  Future<dynamic> navigateTo(String routeName, {dynamic arguments}) {
+    return _navigationKey.currentState
+        .pushNamed(routeName, arguments: arguments);
+  }
+
+  Future<dynamic> navigateAndReplace(String routeName) {
```

Comment:
Would this be useful for the PDF and email situation we have in `pdf_preview.dart`?

Reply:
Yes, I added this just in case we need to do navigation like this.